addons/sale/models/account_invoice.py

```diff
@@ -28,6 +28,8 @@ def _onchange_partner_shipping_id(self):
         """
         Trigger the change of fiscal position when the shipping address is modified.
         """
+        if self.type in self.get_purchase_types():
+            return
```

Comment:
If this onchange is incompatible with the one of `purchase`, this should be using `not in self.get_sale_types()` and something similar should be done in `purchase` with `not in self.get_purchase_types()`